python-lib/dku_utils/node_pool.py

```diff
@@ -17,15 +20,98 @@ def get_node_pool_args(node_pool):
+                'imageBuilder': True # Adding full ECR access to the node group
+            }
+        }
+    }
+    if 'machineType' in node_pool:
+        yaml['instanceType'] = node_pool['machineType']
+
+    if 'diskType' in node_pool:
+        yaml['volumeType'] = node_pool['diskType']
+
+    if 'diskSizeGb' in node_pool and node_pool['diskSizeGb'] > 0:
+        yaml['volumeSize'] = node_pool['diskSizeGb']
+    else:
+        yaml['volumeSize'] = 200 # also defined as default value in parameter-sets/node-pool-request/parameter-set.json
+
+    yaml['desiredCapacity'] = node_pool.get('numNodes', 3)
+    if node_pool.get('numNodesAutoscaling', False):
+        yaml['iam']['withAddonPolicies']['autoScaler'] = True
+        yaml['minSize'] = node_pool.get('minNumNodes', 2)
+        yaml['maxSize'] = node_pool.get('maxNumNodes', 5)
+
+    yaml['tags'] = node_pool.get('tags', {})
+    yaml['taints'] = build_node_pool_taints_yaml(node_pool)
+    node_pool['labels'] = node_pool.get('labels', {})
+    if any(_is_none_or_blank(label_key) for label_key in node_pool['labels'].keys()):
+        logging.error('At least one node pool label key is not valid, please ensure label keys are not empty. Observed labels: [%s]' % ';'.join(node_pool['labels']))
+        raise Exception('At least one node pool label key is not valid, please ensure label keys are not empty. Observed labels: [%s]' % ';'.join(node_pool['labels']))
+    yaml['labels'] = node_pool['labels']
+    yaml['spot'] = node_pool.get('useSpotInstances', False)
+
+    sshPublicKeyName = node_pool.get('publicKeyName', '')
+    if not _is_none_or_blank(sshPublicKeyName):
+        yaml['ssh'] = {
+            'allow': True,
+            'publicKeyName': sshPublicKeyName
+        }
+
+    if networking_settings.get('securityGroups', []):
+        yaml['securityGroups'] = {
+            'attachIDs': map(networking_settings['securityGroups'], lambda security_group: security_group.strip())
+        }
+    yaml['privateNetworking'] = networking_settings.get('privateNetworking', False)
+
+    if node_pool.get('addPreBootstrapCommands', False) and not _is_none_or_blank(node_pool.get('preBootstrapCommands', '')):
+        yaml['preBootstrapCommands'] = yaml.get('preBootstrapCommands', [])
+        yaml['preBootstrapCommands'] += [command.strip()\
+                                          for command in node_pool['preBootstrapCommands'].split('\n')\
+                                              if not _is_none_or_blank(command.strip())]
+
+    return yaml
+
+def build_node_pool_taints_yaml(node_pool):
+    node_pool['taints'] = node_pool.get('taints', [])
+    yaml_taints = []
+    if node_pool['taints']:
+        for taint in node_pool['taints']:
+            if not _is_none_or_blank(taint.get('key', '')):
+                yaml_taints.append({
+                    'key': taint['key'],
+                    'value': taint.get('value', ''),
+                    'effect': taint.get('effect', 'NoSchedule')
```

Comment:
not sure about choosing on behalf of the user... it could result in undesired configuration
I would have raised an exception like for the empty key